src/Beberlei/Bundle/MetricsBundle/DependencyInjection/BeberleiMetricsExtension.php

```diff
@@ -25,13 +26,14 @@ public function load(array $configs, ContainerBuilder $container)
             $container->setDefinition('beberlei_metrics.collector.'.$name, $definition);
         }
 
-        if (!$config['default'] && 1 === count($config['collectors'])) {
-            $container->setAlias('beberlei_metrics.collector', 'beberlei_metrics.collector.'.$name);
-        } elseif ($container->hasDefinition('beberlei_metrics.collector.'.$config['default'])) {
-            $container->setAlias('beberlei_metrics.collector', 'beberlei_metrics.collector.'.$config['default']);
-        } else {
+        if ($config['default'] && $container->hasDefinition('beberlei_metrics.collector.'.$config['default'])) {
+            $name = $config['default'];
+        } elseif (1 !== count($config['collectors'])) {
             throw new \LogicException('You should select a default collector');
         }
+
+        $container->setAlias('beberlei_metrics.collector', 'beberlei_metrics.collector.'.$name);
+        $container->setAlias(Collector::class, 'beberlei_metrics.collector.'.$name);
```

Comment:
The autowiring alias should rather point to `beberlei_metrics.collector` itself

Reply:
done